pkg/workload/workload.go

```diff
@@ -857,6 +857,25 @@ func ApplyAdmissionStatusPatch(ctx context.Context, c client.Client, patch *kueu
 	return c.Status().Patch(ctx, patch, client.Apply, client.FieldOwner(constants.AdmissionName), client.ForceOwnership)
 }
 
+// PatchAdmissionStatus updates the admission status of a workload.
+func PatchAdmissionStatus(ctx context.Context, c client.Client, w *kueue.Workload, strict bool, clk clock.Clock, update func() (bool, error)) error {
+	return PatchAdmissionStatusWithOriginalWorkload(ctx, c, w.DeepCopy(), w, strict, clk, update)
+}
+
+// PatchAdmissionStatusWithOriginalWorkload updates the admission status of a workload.
+// If the WorkloadRequestUseMergePatch feature is enabled, it uses a Merge Patch with update function.
+// Otherwise, it runs the update function and, if updated, applies the SSA Patch status.
+func PatchAdmissionStatusWithOriginalWorkload(ctx context.Context, c client.Client, wOrig *kueue.Workload, w *kueue.Workload, strict bool, clk clock.Clock, update func() (bool, error)) error {
```

Comment:
Why we need this function? I would prefer to inline inside `PatchAdmissionStatus`.